spring-cloud-netflix-eureka-client/src/main/java/org/springframework/cloud/netflix/eureka/serviceregistry/EurekaRegistration.java

```diff
@@ -201,8 +201,4 @@ public int getSecurePort() {
 		return this.instanceConfig.getSecurePort();
 	}
 
-	@Override
-	public void close() throws IOException {
```

Comment:
I worry that this will have negative sure effects: it won't deregister on a normal shutdown.

Reply:
since `DiscoveryClient` is bean itself with marked destroy method, I don't think we need close it twice.
BTW saw exception in logs related this before Egware.3 (when exception was thrown but not muted)

Anyway I've wrote test for this, you may take a look.

Reply:
So where will we shutdown the client otherwise?

Reply:
we don't , spring context will shutdown it for us
```
@Bean(destroyMethod = "shutdown")
@ConditionalOnMissingBean(value = EurekaClient.class, search = SearchStrategy.CURRENT)
public EurekaClient eurekaClient(ApplicationInfoManager manager, EurekaClientConfig config) {
	return new CloudEurekaClient(manager, config, this.optionalArgs, this.context);
}
```
Here is shutdown method `@Bean(destroyMethod = "shutdown")`

Reply:
got it thanks